gammapy/utils/random/inverse_cdf.py

```diff
@@ -77,3 +83,125 @@ def sample(self, size):
 
         index = index + self.random_state.uniform(low=-0.5, high=0.5, size=index.shape)
         return index
+
+
+class MapEventSampler:
+    """Map event sampler.
+
+    Parameters
+    ----------
+
+    npred_map : `~gammapy.maps.Map`
+            Predicted number of counts map.
+    random_state : {int, 'random-seed', 'global-rng', `~numpy.random.RandomState`}
+            Defines random number generator initialisation.
+            Passed to `~gammapy.utils.random.get_random_state`.
+    lc : `~gammapy.time.models.LightCurveTableModel`
+            Input light-curve model of the source, given with columns labelled
+            as "time" (in seconds) and "normalization" (arbitrary units): the bin time
+            HAS to be costant.
+    phase_lc : `~gammapy.time.models.PhaseCurveTableModel`
+            Input phase-curve model of the source, given with columns labelled
+            as "Phase" and "normalization" (arbitrary units): the bin time
+            HAS to be costant.
+    tmin : float
+            Start time of the sampling, given in seconds.
+    tmax : float
+            Stop time of the sampling, given in seconds.
+    """
+
+    def __init__(
+        self, npred_map, random_state=0, lc=None, phase_lc=None, tmin=0, tmax=3600
+    ):
+        self.random_state = get_random_state(random_state)
+        self.npred_map = npred_map
+        self.lc = lc
+        self.phase_lc = phase_lc
+        self.tmin = tmin
+        self.tmax = tmax
+
+    def npred_total(self):
+        """ Calculate the total number of the sampled predicted events.
+            
+        Returns
+        -------
+        random_state.poisson() : int
+                        Number of predicted events.
+        """
+
+        return self.random_state.poisson(np.sum(self.npred_map.data))
+
+    def sample_npred(self):
+        """ Calculate energy and Galactic coordinates of the sampled source events.
+            
+        Returns
+        -------
+        coords : `~numpy.ndarray`
+                Array with coordinates and energies of the sampled events.
+        """
+
+        self.n_events = self.npred_total()
+
+        cdf_sampler = InverseCDFSampler(
+            self.npred_map.data, random_state=self.random_state
+        )
+
+        pix_coords = cdf_sampler.sample(self.n_events)
+        coords = self.npred_map.geom.pix_to_coord(pix_coords[::-1])
+
+        return coords
+
+    def sample_timepred(self):
+        """ Calculate the times of arrival of the sampled source events.
+
+        Returns
+        -------
+        ToA : `~numpy.array`
+            array with times of the sampled events.
+        """
+
+        n_events = self.n_events
+        if (self.lc is not None) and (self.tmax > 0):
```

Comment:
The code for the `lc` and `lc_phase` is to a large degree the same, except that in one case you call `_interpolator(t, ext=3)` and `.evaluate_norm_at_time()` in the other. I think it's probably the better solution
 Maybe just add the `ext` argument to `LightCurveTableModel.evaluate_norm_at_time()`. The everything becomes simpler...

Reply:
I've modified the `~gammapy.time.model.LightCurveTableModel.evaluate_norm_at_time()` including the `ext` argument. By default, this will be equal to 3. I've added also the doc string for this argument.